straxen/plugins/veto_pulse_processing.py

```diff
@@ -72,6 +83,46 @@ def compute(self, raw_records_coin_nv):
         rlinks = strax.record_links(r)
         r = clean_up_empty_records(r, rlinks, only_last=True)
         return r
+    
+    
+def median_baseline(records):
+    """
+    Function which computes the baseline according the pulse's median.
+    
+    :param records: Records 
+    """
+    # Count number of pulses and create buffer:
+    npulses = np.sum(records['record_i'] == 0)
+    
+    if npulses == 1:
+        # This case is simple
+        records = _correct_baseline(records)
+    else:
+        # Now the more complicated case in which we have multiple pulses:
+        _, nextr = strax.record_links(records)
+
+        pulse_i = []
+        for i in np.where(records['record_i'] == 0)[0]:
```

Comment:
Can we do this in a numba loop or vector operations? Perhaps performance is not affected but a while loop in a for loop of 2ns dt records looks ominous.

Reply:
See my comment in the wiki-note